### PR TITLE
SC-4673 CKEditor Update

### PR DESCRIPTION
**Pull request by alriese**
# Description

Update from the CKEditor to version 5.

## Links to Tickets or other pull requests

Ticket: https://ticketsystem.schul-cloud.org/browse/SC-4673
Integration test PR: https://github.com/schul-cloud/integration-tests/pull/99

## New Repos, NPM pakages or vendor scripts

- multiple CKEditor5 related packages

## Screenshots of UI changes

ckeditor5

## Approval for review
- [X] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definition of Done
More and detailed information on the *definition of done* can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 116
- Added 77
           

Complexity increasing per file
==============================
- static/scripts/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 116
- Added 75
           

Complexity increasing per file
==============================
- static/scripts/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 116
- Added 78
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 117
- Added 80
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 117
- Added 65
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 117
- Added 65
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 117
- Added 67
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/filebrowser.js  1
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 70
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/filebrowser.js  1
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 70
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/filebrowser.js  1
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 67
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 67
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 47
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 47
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 46
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 46
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 46
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 48
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 48
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 47
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 46
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 118
- Added 48
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 119
- Added 48
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 119
- Added 45
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 127
- Added 46
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/base.js  -1
+ static/scripts/homework.js  -1
+ static/scripts/news.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 127
- Added 44
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/news.js  -1
+ static/scripts/homework.js  -1
+ static/scripts/base.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 127
- Added 43
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/news.js  -1
+ static/scripts/base.js  -1
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 61
- Added 24
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
+ static/scripts/news.js  -1
+ static/scripts/base.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

File: static/scripts/homework.js [hunk 1]
```diff
@@ -244,26 +184,75 @@ $(document).ready(() => {
...
      * @param file - the new file
      */
-    function addNewUploadedFile(section, file) {
-        let filesCount = section.children().length === 0 ? -1 : section.children().length;
-        let $fileListItem = $(`<li class="list-group-item"><i class="fa fa-file" aria-hidden="true"></i><a href="/files/file?file=${file._id}" target="_blank">${file.name}</a></li>`)
-            .append(`<input type="hidden" name="fileIds[${filesCount + 1}]" value="${file._id}" />`);
+	function addNewUploadedFile(section, file) {
+		const filesCount = section.children().length === 0 ? -1 : section.children().length;
+		const $fileListItem = $(`<li class="list-group-item"><i class="fa fa-file" aria-hidden="true"></i><a href="/files/file?file=${file._id}" target="_blank">${file.name}</a></li>`)
+			.append(`<input type="hidden" name="fileIds[${filesCount + 1}]" value="${file._id}" />`);
```
[the remark this answers is not in the document]
Comment (alriese) [reply static/scripts/homework.js:255]: The input type is hidden, so I think a label is not needed?

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 61
- Added 24
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/ckeditor.js  3
         

Complexity decreasing per file
==============================
+ static/scripts/homework.js  -1
+ static/scripts/news.js  -1
+ static/scripts/base.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 61
- Added 18
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/base.js  -1
+ static/scripts/news.js  -1
+ static/scripts/homework.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Comment by Parnswir** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 82
- Added 10
           

Complexity increasing per file
==============================
- static/scripts/ckeditor/plugins/ckeditor5-helplink/src/helplink.js  1
- static/scripts/ckeditor/ckeditor.js  3
- static/scripts/ckeditor/plugins/ckeditor5-filebrowser/src/filebrowser.js  1
         

Complexity decreasing per file
==============================
+ static/scripts/base.js  -1
+ static/scripts/homework.js  -1
+ static/scripts/news.js  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/schul-cloud/schulcloud-client/pullRequest?prid=5717387&bid=18709010)

**Pull request merged**